spec/services/user_audit_logger_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe UserAuditLogger do
+  describe '#perform' do
+    let(:user_action_event) { create(:user_action_event) }
+    let(:acting_user_verification) { create(:user_verification) }
+    let(:subject_user_verification) { create(:user_verification) }
+    let(:ip_address) { Faker::Internet.ip_v4_address }
+    let(:user_agent) { Faker::Internet.user_agent }
+    let(:config) do
+      {
+        user_action_event: user_action_event,
+        acting_user_verification: acting_user_verification,
+        subject_user_verification: subject_user_verification,
+        status: :initial,
+        ip_address: ip_address,
+        user_agent: user_agent
+      }
+    end
+    let(:logger) { described_class.new(config) }
+
+    it 'creates a user action record' do
+      expect { logger.perform }.to change(UserAction, :count).by(1)
+
+      user_action = UserAction.last
+      expect(user_action).to have_attributes(
+        user_action_event: user_action_event,
+        acting_user_verification: acting_user_verification,
+        subject_user_verification: subject_user_verification,
+        status: 'initial',
+        acting_ip_address: ip_address,
+        acting_user_agent: user_agent
+      )
+    end
+
+    it 'allows setting different status values' do
+      config = {
+        user_action_event: user_action_event,
+        acting_user_verification: acting_user_verification,
+        subject_user_verification: subject_user_verification,
+        status: :error
+      }
+
+      logger = described_class.new(config)
```

Comment:
I don't think you need this line since you defined logger in the `let` above

Reply:
oh I see why you did it this way, hmmm we prob just don't need this spec cause the one above is doing the same thing basically. The unit spec for UserAction model does the actual verification on range of status, so it's prob not important to test all of the 'status' values here